### ISSUE TITLE
reparameterize Teubner-Strey (Trac #472)

### ISSUE THREAD

**Issue by pkienzle** [SasView/sasview#606]
The current implementation of Teubner-Strey use parameters scale, c1, c2 and background.  Since scale and background are automatic parameters in sasmodels, this was changed to a2, c1 and c2.

Propose changing the parameters to the more physically meaningful repeat length d, correlation length xi and a scale factor.  This will give more useful parameter uncertainties.

A little bit of algebra gives a2/c2 and c1/c2 in terms of xi and d:

    a2/c2 = (xi^-2^ + (2 pi/d)^2^)^2^
    c1/c2 = (xi^-2^ - (2 pi/d)^2^)^2^

The scale factor comes from the normalized form (Schubert, 1994):

    I(q) = (8 pi c2 eta^2^ / xi) / (a2 + c1 q^2^ + c2 q^4^)

where

    eta^2^ = phi(1-phi)(delta rho)^2^

for volume fraction phi and scattering length density difference delta rho.

The following parameters should be sufficient:

   sld, sld_solvent, phi in (0,1), xi > 0, and d > 0

using

    I(q) = (8 pi phi(1-phi)(delta rho)^2^) / ((1+(2 pi xi/d)^2^)^2^ + (1-(2 pi xi/d)^2^)^2^ q^2^ + q^4^)

with two of phi, sld and sld_solvent fixed.

Migrated from http://trac.sasview.org/ticket/472
```json
{
    "status": "closed",
    "changetime": "2016-10-01T15:20:07",
    "_ts": "2016-10-01 15:20:07.535192+00:00",
    "description": "The current implementation of Teubner-Strey use parameters scale, c1, c2 and background.  Since scale and background are automatic parameters in sasmodels, this was changed to a2, c1 and c2.\n\nPropose changing the parameters to the more physically meaningful repeat length d, correlation length xi and a scale factor.  This will give more useful parameter uncertainties.\n\nA little bit of algebra gives a2/c2 and c1/c2 in terms of xi and d:\n\n    a2/c2 = (xi^-2^ + (2 pi/d)^2^)^2^\n    c1/c2 = (xi^-2^ - (2 pi/d)^2^)^2^\n\nThe scale factor comes from the normalized form (Schubert, 1994):\n\n    I(q) = (8 pi c2 eta^2^ / xi) / (a2 + c1 q^2^ + c2 q^4^)\n\nwhere\n\n    eta^2^ = phi(1-phi)(delta rho)^2^\n\nfor volume fraction phi and scattering length density difference delta rho.\n\nThe following parameters should be sufficient:\n\n   sld, sld_solvent, phi in (0,1), xi > 0, and d > 0\n\nusing\n\n    I(q) = (8 pi phi(1-phi)(delta rho)^2^) / ((1+(2 pi xi/d)^2^)^2^ + (1-(2 pi xi/d)^2^)^2^ q^2^ + q^4^)\n\nwith two of phi, sld and sld_solvent fixed.",
    "reporter": "pkienzle",
    "cc": "",
    "resolution": "fixed",
    "workpackage": "SasModels Redesign",
    "time": "2015-11-10T00:18:22",
    "component": "SasView",
    "summary": "reparameterize Teubner-Strey",
    "priority": "critical",
    "keywords": "",
    "milestone": "SasView 4.0.0",
    "owner": "ajj",
    "type": "enhancement"
}
```

**Comment by butlerpd** [comment]
Trac update at `2015/11/21 04:38:09`: 
* **butler** changed priority from "minor" to "major"
* **butler** changed workpackage from "SasModels New Model" to "SasModels Redesign"

**Comment by butlerpd** [comment]
Trac update at `2015/11/21 04:44:42`: **butler** changed milestone from "WishList" to "SasView 4.0.0"

**Comment by ajj** [comment]
Trac update at `2016/02/28 11:22:21`: **ajj** changed priority from "major" to "critical"

**Comment by ajj** [comment]
Trac update at `2016/02/28 11:55:17`: 
* **ajj** changed owner from "" to "ajj"
* **ajj** changed status from "new" to "assigned"

**Comment by butlerpd** [comment]
Trac update at `2016/06/20 00:16:42`: **butler** commented:

This is the same request as ticket #663 by Wim Bouwman.  There he is concerned about the instabilities with this parametrization.  Gives reference and attachment for how to change -- should compare against Paul Kienzle's approach above.

**Comment by butlerpd** [comment]
Trac update at `2016/08/16 16:56:30`: 
* **butler** commented:

with the 4.0 release being way late, we clearly don't have the bandwidth to deal with such a change now.  Moving to +1


* **butler** changed milestone from "SasView 4.0.0" to "SasView Next Release +1"
* **butler** changed type from "defect" to "enhancement"

**Comment by butlerpd** [comment]
Trac update at `2016/09/05 02:25:32`: 
* **butler** commented:

Moving back to 4.0 as part of ticket #775 admonition to finalize parameter and model names so as to remain compatible in the future.


* **butler** changed milestone from "SasView Next Release +1" to "SasView 4.0.0"

**Comment by ajj** [comment]
Trac update at `2016/09/27 12:01:13`: 
* **ajj** changed _comment0 from:

> Correct reparameterisation is as per (Schubert, 1994) and (Endo, 2001)
> 
>     a2/c2 = (xi^-2^ + (2 pi/d)^2^)^2^
>     c1/c2 = 2(xi^-2^ - (2 pi/d)^2^) <-- difference is here compared to original ticket text
> 
> The scale factor comes from the normalized form (Schubert, 1994):
> 
>     I(q) = (8 pi c2 eta^2^ / xi) / (a2 + c1 q^2^ + c2 q^4^)
> 
> where
> 
>     eta^2^ = phi(1-phi)(delta rho)^2^
> 
> for volume fraction phi and scattering length density difference delta rho.
> 
> The following parameters should be sufficient:
> 
>    sld, sld_solvent, phi in (0,1), xi > 0, and d > 0
> 
> using
> 
>     I(q) = (8 pi phi(1-phi)(delta rho)^2^) / ((xi^-2^+(2 pi/d)^2^)^2^ + 2(xi^-2^ - (2 pi xi/d)^2^) q^2^ + q^4^)
> 
> with two of phi, sld and sld_solvent fixed.

to:

> 1474977711306779


* **ajj** commented:

Correct reparameterisation is as per (Schubert, 1994) and (Endo, 2001)

    a2/c2 = (xi^-2^ + (2 pi/d)^2^)^2^
    c1/c2 = 2(xi^-2^ - (2 pi/d)^2^) <-- difference is here compared to original ticket text

The scale factor comes from the normalized form (Schubert, 1994):

    I(q) = (8 pi c2 eta^2^ / xi) / (a2 + c1 q^2^ + c2 q^4^)

where

    eta^2^ = phi(1-phi)(delta rho)^2^

for volume fraction phi and scattering length density difference delta rho.

The following parameters should be sufficient:

   sld, sld_solvent, phi in (0,1), xi > 0, and d > 0

using

    I(q) = (8 pi phi(1-phi)(delta rho)^2^) / ((xi^-2^+(2 pi/d)^2^)^2^ + 2(xi^-2^ - (2 pi/d)^2^) q^2^ + q^4^)

with two of phi, sld and sld_solvent fixed.

**Comment by ajj** [comment]
Trac update at `2016/09/30 13:03:21`: 
* **ajj** changed _comment0 from:

> So it seems that we have had various implementations over time:
> 
> 1) Igor + libigor
> 2) sansview + libigor
> 3) a python version 
> 4) a c++ version replacing the python version (ILL 2012)
> 5) version in sasmodels
> 
> in leaving (1) it seems that correct normalisation of parameters was not transferred and so 2->5 are all incorrectly normalised.
> 
> The update with this ticket will aim to match (1), since the citation is from a paper with Steve Kline as an author (along with R Strey)  and he wrote that model, so we will take it as definitive.

to:

> 1475242578317574


* **ajj** commented:

So it seems that we have had various implementations over time:

1) Igor + libigor
2) sansview + libigor
3) a python version 
4) a c++ version replacing the python version (ILL 2012)
5) version in sasmodels

in leaving (1) it seems that correct normalisation of parameters was not transferred and so 2->5 are all "incorrectly" normalised. They all have the correct functional form, but the physical meaning of the parameters and how one extracts the physical meaning varies.

The update with this ticket will aim to match (1), since the citation is from a paper with Steve Kline as an author (along with R Strey)  and he wrote that model, so we will take it as definitive.

**Comment by ajj** [comment]
Trac update at `2016/10/01 15:20:07`: 
* **ajj** commented:

In changeset caddb142198aa5e7856a0850214d72611c3b6ba8:

```text
#!CommitTicketReference repository="sasmodels" revision="caddb142198aa5e7856a0850214d72611c3b6ba8"
Updated and re-parameterised Teubner-Strey model.

Closes #606
Closes #663
```


* **ajj** changed resolution from "" to "fixed"
* **ajj** changed status from "assigned" to "closed"
